hypothesis-python/tests/cover/test_validation.py

```diff
@@ -275,3 +276,27 @@ def test_check_strategy_might_suggest_sampled_from():
     with pytest.raises(InvalidArgument, match="such as st.sampled_from"):
         check_strategy_((1, 2, 3))
     check_strategy_(integers(), "passes for our custom coverage check")
+
+
+@pytest.mark.parametrize("codec", ["ascii", "utf-8"])
+def test_warn_on_strings_matching_common_codecs(codec):
+    with pytest.warns(
+        HypothesisWarning,
+        match=f"it seems like you are trying to use the codec {codec!r}",
+    ):
+
+        @given(st.text(codec))
+        def f(s):
+            pass
+
+        f()
+
+    # if we reorder, it doesn't warn anymore
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
I will be _so happy_ when 3.10 goes EOL and I can use [`with catch_warnings(action="error"):`](https://docs.python.org/3/library/warnings.html#warnings.catch_warnings)... the ~six-year lag between CPython contributions and getting to use them in oss is so painful sometimes.